src/overlays/actors/ovl_En_Go/z_en_go.h

```diff
@@ -2,105 +2,127 @@
-    /* 0x190 */ EnGoActionFunc unk_190;
+    /* 0x18C */ EnGoActionFunc interruptedActionFunc;
+    /* 0x190 */ EnGoActionFunc graveBroDialogActionFunc;
     /* 0x194 */ ColliderCylinder colliderCylinder;
     /* 0x1E0 */ UNK_TYPE1 unk1E0[0x4C];
     /* 0x22C */ ColliderSphere colliderSphere;
-    /* 0x284 */ Path* unk_284;
-    /* 0x288 */ s8 unk_288;
-    /* 0x289 */ s8 unk_289;
-    /* 0x28C */ s32 unk_28C;
-    /* 0x290 */ Vec3f unk_290;
-    /* 0x29C */ Vec3f unk_29C;
-    /* 0x2A8 */ Vec3s unk_2A8;
-    /* 0x2AE */ Vec3s unk_2AE;
-    /* 0x2B4 */ Vec3s jointTable[18];
-    /* 0x320 */ Vec3s morphTable[18];
-    /* 0x38C */ Actor* unk_38C;
-    /* 0x390 */ u16 unk_390;
-    /* 0x392 */ u16 unk_392;
-    /* 0x394 */ u8 unk_394;
-    /* 0x398 */ f32 unk_398;
-    /* 0x39C */ f32 unk_39C;
-    /* 0x3A0 */ f32 unk_3A0;
-    /* 0x3A4 */ f32 unk_3A4;
-    /* 0x3A8 */ f32 unk_3A8;
+    /* 0x284 */ Path* gatekeeperPath;
+    /* 0x288 */ s8 objIndexTaisou;
+    /* 0x289 */ s8 objIndexHakuginDemo;
+    /* 0x28C */ s32 msgScriptResumePos;
+    /* 0x290 */ Vec3f headPos;
+    /* 0x29C */ Vec3f bodyPos;
+    /* 0x2A8 */ Vec3s headRot;
+    /* 0x2AE */ Vec3s bodyRot;
+    /* 0x2B4 */ Vec3s jointTable[GORON_LIMB_MAX];
+    /* 0x320 */ Vec3s morphTable[GORON_LIMB_MAX];
+    /* 0x38C */ Actor* attentionTarget;
+    /* 0x390 */ u16 actionFlags;
+    /* 0x392 */ u16 lastTextId;
+    /* 0x394 */ u8 springArrivalCurrentCsAction;
+    /* 0x398 */ f32 currAnimPlaySpeed;
+    /* 0x39C */ f32 iceBlockScale;
+    /* 0x3A0 */ f32 iceBlockAlpha;
+    /* 0x3A4 */ f32 scaleFactor;
+    /* 0x3A8 */ f32 scaleFactorPoundDistortion;
     /* 0x3AC */ UNK_TYPE1 unk3AC[0x2];
-    /* 0x3AE */ s16 unk_3AE;
-    /* 0x3B0 */ s16 unk_3B0;
-    /* 0x3B2 */ s16 unk_3B2;
-    /* 0x3B4 */ s16 unk_3B4;
-    /* 0x3B6 */ s16 unk_3B6;
-    /* 0x3B8 */ s16 unk_3B8;
-    /* 0x3BA */ s16 unk_3BA;
-    /* 0x3BC */ s16 unk_3BC;
-    /* 0x3BE */ s16 unk_3BE;
-    /* 0x3C0 */ s16 unk_3C0;
-    /* 0x3C2 */ s16 unk_3C2;
-    /* 0x3C4 */ s16 unk_3C4;
-    /* 0x3C6 */ s16 unk_3C6;
-    /* 0x3C8 */ s16 unk_3C8[3];
-    /* 0x3CE */ s16 unk_3CE[3];
-    /* 0x3D4 */ s16 unk_3D4;
-    /* 0x3D8 */ void* unk_3D8;
-    /* 0x3DC */ s32 unk_3DC;
+    /* 0x3AE */ union {
+        s16 snorePhase;      // Used whilst a Goron is sleeping, +1024 at a time
+        s16 elapsedHangtime; // Hangtime at the top of a pound.
+    };
+    /* 0x3B0 */ s16 headRotZ;
+    /* 0x3B2 */ s16 headRotY;
+    /* 0x3B4 */ s16 bodyRotZ;
+    /* 0x3B6 */ s16 bodyRotY;
+    /* 0x3B8 */ s16 indexCutscene;
+    /* 0x3BA */ s16 loseAttentionTimer;
+    /* 0x3BC */ s16 eyeTimer;
+    /* 0x3BE */ s16 eyeTexIndex;
+    /* 0x3C0 */ s16 cutsceneState;
+    /* 0x3C2 */ s16 cutsceneDelayTimer;
+    /* 0x3C4 */ s16 gatekeeperAnimState;
+    /* 0x3C6 */ s16 harmlessTimer;
+    /* 0x3C8 */ s16 limbRotTableZ[3];
+    /* 0x3CE */ s16 limbRotTableY[3];
+    /* 0x3D4 */ s16 surprisePhase;
+    /* 0x3D8 */ void* msgEventCb;
```

Comment:
This has type `s32 (*)(Actor*, PlayState*)`, just make `typedef s32 (*MsgEventFunc)(Actor*, PlayState*)` next to the actionfunc one and use it here.